### PR TITLE
telegraf.conf: add SSL support and short hostname

### PR DESCRIPTION
**Pull request by skdst**
added checkboxes in web gui for telegraf.conf options
  hostname (Use short hostname instead of FQDN)
  insecure_skip_verify (Use SSL but skip chain and host verification)
added support for setting CA certificate in telegraf.conf

### CODE REVIEW

File: net-mgmt/pfSense-pkg-Telegraf/files/usr/local/pkg/telegraf.inc [hunk 1]
```diff
@@ -51,12 +51,29 @@ function telegraf_resync_config() {
...
 	}
 
+	$shortname='';
+	if ($telegraf_conf['shortname'] == 'on') {
+		$shortname="\thostname = \"" . $config['system']['hostname'] . "\"";
+	}
+
+	$ca_pem_file = "/usr/local/etc/telegraf.ca";
+	$a_ca=$config['ca'];
+	if (!is_array($a_ca)) $a_ca=array();
```
Comment (jim-p) [inline net-mgmt/pfSense-pkg-Telegraf/files/usr/local/pkg/telegraf.inc]: Please use [K&R, BSD KNF variant style](https://doc.pfsense.org/index.php/Developer_Style_Guide#Indent_style), even for simple statements, to keep them consistent and easy to read, and be careful to use matching and proper indentation.

File: net-mgmt/pfSense-pkg-Telegraf/files/usr/local/pkg/telegraf.inc [hunk 2]
```diff
@@ -51,12 +51,29 @@ function telegraf_resync_config() {
...
+	if (count($a_ca)) {
+		$ca_pem='';
+		foreach ($a_ca as $ca) {
+			$ca_pem .= base64_decode($ca['crt']);
+		}
+		 file_put_contents($ca_pem_file, $ca_pem);
```
Comment (jim-p) [inline net-mgmt/pfSense-pkg-Telegraf/files/usr/local/pkg/telegraf.inc]: Remove the extra space here before `file_put_contents`.

**Comment by skdst** [comment]
@rbgarga @jim-p thanks for your reviews, please verify again.

File: net-mgmt/pfSense-pkg-Telegraf/files/usr/local/pkg/telegraf.inc [hunk 3]
```diff
@@ -51,12 +51,31 @@ function telegraf_resync_config() {
 		$telegraf_conf['interval'] = 10;
 	}
 
+	$shortname = '';
+	if ($telegraf_conf['shortname'] == 'on') {
+		$shortname = "\thostname = \"" . $config['system']['hostname'] . "\"";
+	}
+
+	$ca_pem_file = "/usr/local/etc/telegraf.ca";
+	$a_ca = $config['ca'];
```
Comment (rbgarga) [inline net-mgmt/pfSense-pkg-Telegraf/files/usr/local/pkg/telegraf.inc]: There is no need to make a copy of `$config['ca']` just to use it on below foreach(). It's better to just use it instead

Comment (skdst) [reply net-mgmt/pfSense-pkg-Telegraf/files/usr/local/pkg/telegraf.inc]: I adapted the code style from official pfsense source code like you see in the following two references:

https://github.com/pfsense/pfsense/blob/7c41a378ee282fb43b285b66283160686e7cb5ca/src/usr/local/www/vpn_openvpn_client.php#L51

https://github.com/pfsense/pfsense/blob/7c41a378ee282fb43b285b66283160686e7cb5ca/src/usr/local/www/vpn_openvpn_client.php#L685

**Comment by skdst** [comment]
Please review again

**Comment by rbgarga** [comment]
@skdst changes are now approved. Please rebase your fork to resolve conflicts and we will get it merged

**Pull request merged**